### PR TITLE
Allow running multiple checks

### PR DESCRIPTION
**Pull request by fbertsch**
This has been very useful for checking e.g. all `baseline_clients_first_seen` tables for errors.

Another nice improvement would be parallelization, but that is not required.
Also added some typing.

Checklist for reviewer:

- [x] Commits should reference a bug or github issue, if relevant (if a bug is referenced, the pull request should include the bug number in the title).
- [x] If the PR comes from a fork, trigger integration CI tests by running the [Push to upstream workflow](https://github.com/mozilla/bigquery-etl/actions/workflows/push-to-upstream.yml) and provide the `<username>:<branch>` of the fork as parameter. The parameter will also show up
in the logs of the `manual-trigger-required-for-fork` CI task together with more detailed instructions.
- [x] If adding a new field to a query, ensure that the schema and dependent downstream schemas have been updated.
- [x] When adding a new derived dataset, ensure that data is not available already (fully or partially) and recommend extending an existing dataset in favor of creating new ones. Data can be available in the [bigquery-etl repository](https://github.com/mozilla/bigquery-etl), [looker-hub](https://github.com/mozilla/looker-hub) or in [looker-spoke-default](https://github.com/mozilla/looker-spoke-default/tree/e1315853507fc1ac6e78d252d53dc8df5f5f322b).

For modifications to schemas in restricted namespaces (see [`CODEOWNERS`](https://github.com/mozilla/bigquery-etl/blob/main/CODEOWNERS)):
- [ ] Follow the [change control procedure](https://docs.google.com/document/d/1TTJi4ht7NuzX6BPG_KTr6omaZg70cEpxe9xlpfnHj9k/edit#heading=h.ttegrcfy18ck)

### CODE REVIEW

File: bigquery_etl/cli/utils.py [hunk 1]
```diff
@@ -78,29 +79,31 @@ def table_matches_patterns(pattern, invert, table):
...
+        yield checks_file, project, dataset, table
     else:
         print(f"No checks.sql file found in {sql_path}/{project_id}/{pattern}")
         return None, None, None, None
 
 
 def paths_matching_name_pattern(
     pattern, sql_path, project_id, files=["*.sql"], file_regex=QUERY_FILE_RE
-):
+) -> List[Path]:
```
Comment (fbertsch) [inline bigquery_etl/cli/utils.py:104]: I wanted typing in the above function, and `mypy` complained unless I added it here as well.

**Comment by alekhyamoz** [comment]
@fbertsch  can you please help me understand how this differs from having all the checks in a single file?

**Comment by fbertsch** [comment]
> @fbertsch can you please help me understand how this differs from having all the checks in a single file?

@alekhyamoz this is for running checks across multiple tables. For example, when working with the `glean_usage` tables, we generate checks on many apps at once. Instead of me having to manually run checks for each individually, it was much easier to run `bqetl check run *.baseline_clients_first_seen_v1`

**Review by alekhyamoz — APPROVED** [review]
looks good to me! Thank you for working on adding this feature :)

**Pull request merged**